File: tl/src/main/java/com/github/badoualy/telegram/tl/api/requests/TLRequestAuthCheckPhone.java

package com.github.badoualy.telegram.tl.api.requests;

import com.github.badoualy.telegram.tl.core.*;
import com.github.badoualy.telegram.tl.*;
import java.io.*;
import rx.Observable;

import static com.github.badoualy.telegram.tl.StreamUtils.*;


public class TLRequestAuthCheckPhone extends TLMethod<com.github.badoualy.telegram.tl.api.auth.TLCheckedPhone> {

    public static final int CLASS_ID = 0x6fe51dfb;

    public int getClassId() {
        return CLASS_ID;
    }


    public TLRequestAuthCheckPhone(        String _phoneNumber) {
        this.phoneNumber = _phoneNumber;

    }



    public com.github.badoualy.telegram.tl.api.auth.TLCheckedPhone deserializeResponse(InputStream stream, TLContext context) throws IOException {

        TLObject res = readTLObject(stream, context);
        if (res == null) {
            throw new IOException("Unable to parse response");
        }
        if (res instanceof com.github.badoualy.telegram.tl.api.auth.TLCheckedPhone) {
            return (com.github.badoualy.telegram.tl.api.auth.TLCheckedPhone)res;
        }
        else {
            throw new IOException("Incorrect response type. Expected com.github.badoualy.telegram.tl.api.auth.TLCheckedPhone, got: " + res.getClass().getCanonicalName());
        }

    }
        


    protected String phoneNumber;


    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String value) {
        this.phoneNumber = value;
    }


    @Override
    public void serializeBody(OutputStream stream) throws IOException {

        writeTLString(this.phoneNumber, stream);
    }


    @Override
    public void deserializeBody(InputStream stream, TLContext context) throws IOException {

        this.phoneNumber = readTLString(stream);
    }



    @Override
    public String toString() {
        return "auth.checkPhone#6fe51dfb";
    }

}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/request/TLRequestAuthCheckPhone.kt
package com.github.badoualy.telegram.tl.api.request

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import com.github.badoualy.telegram.tl.api.auth.TLCheckedPhone
import com.github.badoualy.telegram.tl.core.TLMethod
import com.github.badoualy.telegram.tl.serialization.TLDeserializer
import com.github.badoualy.telegram.tl.serialization.TLSerializer
import java.io.IOException

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLRequestAuthCheckPhone() : TLMethod<TLCheckedPhone>() {
    var phoneNumber: String = ""

    private val _constructor: String = "auth.checkPhone#6fe51dfb"

    override val constructorId: Int = CONSTRUCTOR_ID

    constructor(phoneNumber: String) : this() {
        this.phoneNumber = phoneNumber
    }

    @Throws(IOException::class)
    override fun deserializeResponse(tlDeserializer: TLDeserializer): TLCheckedPhone = tlDeserializer.readTLObject(TLCheckedPhone::class, TLCheckedPhone.CONSTRUCTOR_ID)

    @Throws(IOException::class)
    override fun serializeBody(tlSerializer: TLSerializer) = with (tlSerializer)  {
        writeString(phoneNumber)
    }

    @Throws(IOException::class)
    override fun deserializeBody(tlDeserializer: TLDeserializer) = with (tlDeserializer)  {
        phoneNumber = readString()
    }

    override fun computeSerializedSize(): Int {
        var size = SIZE_CONSTRUCTOR_ID
        size += computeTLStringSerializedSize(phoneNumber)
        return size
    }

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLRequestAuthCheckPhone) return false
        if (other === this) return true

        return phoneNumber == other.phoneNumber
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x6fe51dfb.toInt()
    }
}
